docs/source/reference/logging.rst

```diff
@@ -0,0 +1,197 @@
+Logging
+=======
+
+To log with Perceval you can either use our logger or the python one. By default, our logger will be used.
+
+To have the perceval message log with the python logger, use this method:
+
+.. code-block:: python
+
+    import perceval
+    perceval.utils.use_python_logger()
+
+To use back the perceval logger use this method:
+
+.. code-block:: python
+
+    import perceval
+    perceval.utils.use_perceval_logger()
+
+.. note:: If you use the python logger, use directly the module logging of python to configure it (except for channel configuration)
+
+This module defines functions and classes which implement a flexible event logging system for any Perceval script.
+
+It is build to work similarly as the python logging module.
+
+Logger
+------
+
+The logger is instantiate at perceval initialization.
+
+In order to use it you'll just need to import it:
+
+.. code-block:: python
+
+    from perceval.utils import logger
+
+To log message, you can use it the same way as the python logger:
+
+.. code-block:: python
+
+    from perceval.utils import logger
+    logger.info('I log something as info')
+
+
+Saving log in file
+^^^^^^^^^^^^^^^^^^
+
+By default the log are not save in a file.
+If this feature is activated, the log file will be in the perceval persistent data folder and the path of the file will be printed at the beginning of your perceval script.
+
+If order to activate / deactivate the logging in a file you can use the following methods:
+
+.. code-block:: python
+
+    from perceval.utils import logger
+    logger.enable_file()
+    logger.disable_file()
+
+Levels
+------
+
+You can use the logger to log message at different level. Each level represent a different type of message.
+
+The level are listed by ascending important in the following table.
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :width: 100%
+   :align: center
+
+   * - Log level
+     - Perceval call
+     - Usage
+   * - DEBUG
+     - ``logger.debug``
+     - Detailed information, typically of interest only when diagnosing problems.
+   * - INFO
+     - ``logger.info``
+     - Confirmation that things are working as expected.
+   * - WARNING
+     - ``logger.warn``
+     - An indication that something unexpected happened, or indicative of some problem in the near future (e.g. ‘disk space low’). The software is still working as expected.
+   * - ERROR
+     - ``logger.error``
```

Comment:
It seems we called it "err" in exqalibur and not error :/

Reply:
Yes but the python logger module call it error. I think it's a good idea to keep the same interface :)

Reply:
Yes but the python logger module call it error. I think it's a good idea to keep the same interface :)

Reply:
What I meant is: we use `exqalibur.logger.level` enum in Python, and exqalibur exposes the error level as `exqalibur.logger.level.err` which is exposed by Python as `perceval.utils.logging.level.err`

So isn't there an issue with the doc?